packages/nimble-components/src/rich-text/base/index.ts

```diff
@@ -36,34 +55,50 @@ export abstract class RichText extends FoundationElement {
      * @internal
      */
     public handleChange(source: unknown, args: unknown): void {
-        if (
-            source instanceof MentionInternals
-            && MarkdownParserMentionConfiguration.isObservedMentionInternalsProperty(
+        if (source instanceof MentionInternals && typeof args === 'string') {
+            this.richTextUpdateTracker.trackMentionInternalsPropertyChanged(
                 args
-            )
-        ) {
-            this.configuration = this.createConfig();
+            );
         }
     }
 
-    protected mentionElementsChanged(_old: unknown, _new: unknown): void {
-        this.observeMentionInternals();
-        this.configuration = this.createConfig();
+    /**
+     * @internal
+     */
+    public createConfig(): void {
+        this.validate();
+        if (this.richTextValidator.isValid()) {
+            if (
+                this.richTextUpdateTracker.updateMappingConfigs
+                || this.richTextUpdateTracker.updatePattern
+            ) {
+                this.configuration = new Configuration(this.mentionElements);
+            }
+        } else {
+            this.configuration = undefined;
+        }
     }
 
-    protected createConfig(): Configuration {
-        return new Configuration(this.mentionElements);
+    protected validate(): void {
+        this.richTextValidator.validate(this.mentionElements);
     }
 
-    private childItemsChanged(_prev: unknown, _next: unknown): void {
-        void this.updateMentionElementsFromChildItems();
+    private childItemsChanged(
+        prev: Element[] | undefined,
+        next: Element[]
+    ): void {
+        if (prev?.length || next.length) {
+            void this.updateMentionElementsFromChildItems();
+        }
```

Comment:
This seems fishy to me, if the mention elements are cleared then the rest of the system will ignore that change?

If the mention elements are removed we should NOT hold onto the old mention config. That means the element configuration does not map to what the element renders.

Reply:
The following should behave the exact same:

```
<rich-text>
```


and 

```
<rich-text> --> <rich-text><mention></rich-text> --> <rich-text>
```

If we ignore the mention config when it drops to zero then the above to will behave differently. That is wrong

Reply:
Oh wait, maybe I was confused. What is the check guarding against exactly? is it just the transition from `prev = undefined` to `next = empty array`?

If it is just that transition we are ignoreing can you write it as `if (prev !== undefined)`? I think that better conveys we are just ignoring the initialization.

Reply:
@aagash-ni @vivinkrishna-ni there was offline requests to have this PR reviewed, so FYI it was reviewed yesterday and is waiting for your follow up

Reply:
Sorry for the delayed response. 

The reason for the check is that some of the async tests fail with `if (prev !== undefined)` and here is the [pipeline failure build](https://github.com/ni/nimble/actions/runs/7472203971/job/20333903895). This is because in some of the transitions, the `updateMentionElementsFromChildItems()` is called which has the async operation of waiting until the child items are defined and resolve promises. So I came up with the below observations of when it is required to update the content in the rich text components:

![image](https://github.com/ni/nimble/assets/123377523/6669e405-b6e3-40df-b5b5-ad73a3b0f420)

1. In the above image, the red-marked transition is the one that makes the mentioned test fail as we are using `waitUntilCustomElementsDefinedAsync()` operation within the update method. Also, in this transition, we don't want to update the mention elements, as the `childItems` is empty, so `prev?.length || next.length` condition guards the `prev === []` and `next === []`.
2. Also, the yellow marked record is expected to call the `updateMentionElementsFromChildItems()` but this is not an issue for this use case as we are initializing the `childItems` with an empty array during the initialization.

Reply:
Your response describes the behavior being observed, i.e it causes a test failure, but that doesn't tell me if the behavior makes sense and there is a valid reason to skip the cases, if there is a problem with how the test is written, or if skipping the cases is masking an underlying problem. As an FYI that means I need to spend more time understanding the code and why the change makes sense and that will take longer to review. 

> Also, in this transition, we don't want to update the mention elements, as the childItems is transitioned from empty to empty

That doesn't tell me why it makes sense for that to cause some error or test failure. It seems like a strong indication that something else is incorrect because it is surprising that causes an issue

Reply:
In the test cases, after the `await connect()`, the `waitUntilCustomElementsDefinedAsync()` is called. During that time, the element is not initialized and it looks like the below:
![image](https://github.com/ni/nimble/assets/123377523/32f546af-d94d-4840-91bc-78ea3e09dcb2)

The breakpoint is here after the connect call for the above case:
![image](https://github.com/ni/nimble/assets/123377523/1273aab2-e828-4dec-be60-cf86f03f4c79)

I am exactly not sure of the reason and as mentioned in [this thread](https://github.com/ni/nimble/pull/1099/files#r1130001024) for the table batch update changes, once adding the `await waitForUpdatesAsync()` after the `await connect()`, the **test passes**. So I have updated the same and set the condition to `prev !== undefined`. So I think this test failure might be related to the test environments because of the async update.

Reply:
That seems aligned with behaviors seen in the table. Thanks for the investigation.
Though like seen with the table and in this PR you may need to expect that some timings in tests will change, I assume that may impact the Skyline tests. I'm assuming the page objects are being re-used in Skyline. You may want to consider if the timing changes can be encapsulated in the page objects.

Reply:
I think currently the Skyline tests are not using the page objects and there are necessary awaits already in place in the component tests: [@mention test block in sl-comments ](https://dev.azure.com/ni/DevCentral/_git/Skyline?path=/Web/Workspaces/SystemLinkShared/projects/systemlink-comments-lib/src/components/sl-comment/sl-comment.component.spec.ts&version=GC6aa2ab9ea824901f40c045b2abb90f99f702b11d&line=930&lineEnd=931&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents). For now, I think we may not want to add timing changes to the page objects for encapsulation, but if needed, we will update it in a separate PR.

I am completing this PR now and if you have any concerns or any other test failures we encounter in Skyline we will create a follow-up PR to address it.